### PR TITLE
add custom nonce flag to cmd package token subcommand

### PR DESCRIPTION
**Pull request by Ruide**
- add a custom-nonce flag to allow user supplied string array for Custom Nonce
- added the flag in token subcommand for generating tokens with the custom nonce field
- added the flag in unit test

### CODE REVIEW

File: cmd/token.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ The OIDC token includes claims regarding the GCE VM, which is verified by Attest
 			Challenge:      challenge,
 			GcpCredentials: principalTokens,
 			Attestation:    attestation,
-			TokenOptions:   verifier.TokenOptions{CustomAudience: audience, TokenType: "OIDC"},
+			TokenOptions:   verifier.TokenOptions{CustomAudience: audience, CustomNonce: customNonce, TokenType: "OIDC"},
```
Comment (JoshuaKrstic) [inline cmd/token.go:141]: Does this mean that default token will have this custom nonce every time? What is this used for?

Comment (Ruide) [reply cmd/token.go:141]: The default value for `customNonce` is nil. Thus, the default token would not contain custom nonce. This is used for on-demand attestation for CVMs for customer to build upon.

Comment (JoshuaKrstic) [reply cmd/token.go:141]: Ah yes sorry, totally missed the directory this was in.

**Review by alexmwu — CHANGES_REQUESTED** [review]
Please add a test that uses custom nonce and the fake verifier

**Review by JoshuaKrstic — APPROVED** [review]

File: cmd/flags.go [hunk 2]
```diff
@@ -144,6 +145,12 @@ func addAudienceFlag(cmd *cobra.Command) {
 		"the audience field in the claims token. Cannot be sts.googleapis.com.")
 }
 
+// Lets this command specify custom nonce field of the attestation token.
+func addCustomNonceFlag(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringArrayVar(&customNonce, "custom-nonce", nil,
+		"the custom nonce field in the claims token. use this flag multiple times to add multiple custom nonce.")
```
Comment (alexmwu) [inline cmd/flags.go:151]: "the custom nonce field in the claims token. Use this flag multiple times to add multiple custom nonces."

File: verifier/util/fake_attestation_server.go [hunk 3]
```diff
@@ -71,3 +88,27 @@ func NewMockAttestationServer() (*MockAttestationServer, error) {
...
+	}
+	defer r.Body.Close()
+
+	if err := protojson.Unmarshal(body, req); err != nil {
+		return fmt.Errorf("error decoding attestation request: %v", err)
+	}
+
+	if req.TokenOptions.Nonce != nil || req.TokenOptions.Audience != "" {
+		if req.TokenOptions.Nonce[0] != FakeCustomNonce[0] || req.TokenOptions.Nonce[1] != FakeCustomNonce[1] {
+			return fmt.Errorf("error comparing custom nonce: %v", err)
```
Comment (alexmwu) [inline verifier/util/fake_attestation_server.go:107]: what about failure mode tests?

**Review by alexmwu — APPROVED** [review]

**Pull request merged**